### PR TITLE
fix: explat client experiment name validation to match server side

### PR DESCRIPTION
**Pull request by rjchow**
Related to https://github.com/woocommerce/woocommerce/issues/41854

## Proposed Changes

* Adjusted the experiment name validation in explat client to match the actual server side validation

## Testing Instructions

* Analysis of the code and unit test should be sufficient

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/explat-client/src/internal/validations.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export function isObject( x: unknown ): x is Record< string, unknown > {
  * @param name The data to test
  */
 export function isName( name: unknown ): name is string {
-	return typeof name === 'string' && name !== '';
+	return typeof name === 'string' && name !== '' && /^[a-z0-9_,]*$/.test( name );
```
Comment (jessie-ross) [inline packages/explat-client/src/internal/validations.ts]:
```suggestion
	return typeof name === 'string' && name !== '' && /^[a-z0-9_]*$/.test( name );
```

Seems like the comma is a bug?

**Review by jessie-ross — APPROVED** [review]
Also bump the version number here and in `explat-client-react-helpers`, add an entry in the changelog. See the bottom of `/docs/monorepo.md` for release instructions.

It will need to be updated downstream as well, creating a P2 post on better experiments to announce, copying this post:
pbmo2S-27o-p2

File: packages/explat-client/src/internal/validations.ts (hunk 1 above)
Comment (rjchow) [reply packages/explat-client/src/internal/validations.ts]: Thanks for catching that! I did think it looked a bit weird but neglected to double check

**Review by chihsuan — APPROVED** [review]
LGTM! 🚢

**Pull request merged**